Source/WebCore/platform/audio/SharedAudioDestination.cpp

```diff
@@ -124,91 +122,85 @@ SharedAudioDestinationAdapter::~SharedAudioDestinationAdapter()
     auto key = std::make_tuple(m_numberOfOutputChannels, m_sampleRate);
     sharedMap().remove(key);
     protectedDestination()->clearCallback();
+    callAllConfigurationHandlers(false);
 }
 
 void SharedAudioDestinationAdapter::addRenderer(SharedAudioDestination& renderer, CompletionHandler<void(bool)>&& completionHandler)
 {
-    protectedConfigurationQueue()->dispatch([this, weakThis = ThreadSafeWeakPtr { *this }, renderer = RefPtr { &renderer }, completionHandler = WTFMove(completionHandler)] () mutable {
-        RefPtr protectedThis = weakThis.get();
-        if (!protectedThis) {
-            callOnMainThread([completionHandler = WTFMove(completionHandler)] () mutable { completionHandler(false);
-            });
-            return;
-        }
-
-        if (!m_renderers.contains(renderer))
-            m_renderers.append(WTFMove(renderer));
-        configureRenderThread(WTFMove(completionHandler));
-    });
+    if (!m_renderers.contains(&renderer))
+        m_renderers.append(&renderer);
+    configureRenderThread(WTFMove(completionHandler));
 }
 
 void SharedAudioDestinationAdapter::removeRenderer(SharedAudioDestination& renderer, CompletionHandler<void(bool)>&& completionHandler)
 {
-    protectedConfigurationQueue()->dispatch([this, weakThis = ThreadSafeWeakPtr { *this }, renderer = RefPtr { &renderer }, completionHandler = WTFMove(completionHandler)] () mutable {
-        RefPtr protectedThis = weakThis.get();
-        if (!protectedThis) {
-            callOnMainThread([completionHandler = WTFMove(completionHandler)] () mutable { completionHandler(false);
-            });
-            return;
-        }
-
-        m_renderers.removeFirst(renderer);
-        configureRenderThread(WTFMove(completionHandler));
-    });
+    m_renderers.removeFirst(&renderer);
```

Comment:
nit: maybe we can `ASSERT(!m_renderers.contains(&renderer));` after that, to be safe.